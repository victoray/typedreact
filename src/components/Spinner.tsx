import React from "react";


/**
 * Displays a spinning loader during data loading.
 */
const Spinner = ({message = "Loading...."}) => {

    return (
        <div className="ui active dimmer">
            <div className="ui big text loader"> {message}</div>
        </div>
    );
};

export default Spinner;
