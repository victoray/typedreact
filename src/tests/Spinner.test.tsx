import React from "react";
import renderer from "react-test-renderer";
import Spinner from "../components/Spinner";


it("should render loading spinner", () => {
    const tree = renderer
        .create(<Spinner/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
