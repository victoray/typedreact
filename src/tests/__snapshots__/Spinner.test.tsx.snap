// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render loading spinner 1`] = `
<div
  className="ui active dimmer"
>
  <div
    className="ui big text loader"
  >
     
    Loading....
  </div>
</div>
`;
